dev-client/components/SiteMap.tsx

```diff
@@ -0,0 +1,47 @@
+import Mapbox, {Camera, MarkerView, UserLocation} from '@rnmapbox/maps';
+import React, {useEffect, useRef} from 'react';
+import {type DisplaySite} from '../datatypes/sites';
+import {type Position} from '@rnmapbox/maps/lib/typescript/types/Position';
+import {PermissionsAndroid} from 'react-native';
+import {IconButton, Icon} from 'native-base';
+// TODO: This vector icons module appears to be missing types
+import VIcon from 'react-native-vector-icons/MaterialIcons';
+
+type SiteMapProps = {
+  sites: DisplaySite[];
+  center: Position;
+};
+
+// REVIEW: This is a bad name, any suggestions?
+export default function SiteMap({sites, center}: SiteMapProps): JSX.Element {
+  const camera = useRef<Camera>(null);
+
+  useEffect(() => {
+    camera.current?.setCamera({
+      centerCoordinate: center,
+    });
+  }, [center]);
+
+  return (
+    <Mapbox.MapView
+      style={{
+        flex: 1,
+      }}>
+      <Camera ref={camera} />
+      <UserLocation />
+      {sites.map(site => {
+        return (
+          <MarkerView coordinate={[site.lon, site.lat]} key={site.key}>
+            <IconButton
+              icon={<Icon as={VIcon} name="location-on" />}
```

Comment:
This is somewhat confusing. I don't worry about it here, but I think it would be nice to have a component to deal with this stuff for us, because we will be doing it a lot.

What this code is doing:
- `IconButton` is a nativebase component
- It uses `Icon` as the icon component, another nativebase component. The main advantage here is that we can pass our theme values to the icon.
- Finally, `VIcon` is the `react-native-vector-icon` component. It takes the `name` prop and displays the icon we want to display.

I think actually I should add the info about how to pick out the icon to the README.

Reply:
suggestion: could we create a very thin abstraction over this and document what it's doing there? then our calls can be a bit less repetitive if they look like `<MyIconButton icon="location-on" size="sm" ... />

Reply:
Yes, I ended up doing this in my next PR because it was getting annoying.